service/src/main/java/com/epam/esm/servise/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+        }
+
+    }
+
+    @Transactional
+    @Override
+    public void updateEntity(long id, User user) {
+        Optional<User> user1 = repository.getEntityById(id);
+        if (user1.isPresent()) {
+            user.setId(id);
+            if(user.getNickName()==null)
+                user.setNickName(user1.get().getNickName());
+            if(user.getEmail()==null)
+                user.setEmail(user1.get().getEmail());
+            repository.updateEntity(user);
+        } else {
+            throw new NoSuchEntityException(languageMassage.getMessage("message.user.with.id"));
+        }
+    }
+
+    @Transactional
+    @Override
+    public Optional<UserModel> getEntity(long id) {
+        Optional<User> user = Optional.ofNullable(repository.getEntityById(id)).orElseThrow();
+        if (user.isEmpty()) {
+            throw new NoSuchEntityException(languageMassage.getMessage("message.user.with.id"));
+        }
+        return user.map(readMapper::mapFrom);
+    }
+
+    @Transactional
+    @Override
+    public void deleteEntity(long id) {
+        Optional<User> user = repository.getEntityById(id);
+        if (user.isEmpty()) {
+            throw new NoSuchEntityException(languageMassage.getMessage("message.user.with.id"));
+        } else {
+            repository.deleteEntity(id);
+        }
+    }
+
+    @Transactional
+    public int countAllUsers() {
+        return repository.countAllUsers();
+    }
+
+    @Transactional
+    public Optional<ReadUserModel> getUserByName(String name) {
+        Optional<User> user = Optional.ofNullable(repository.getUserByName(name)).orElseThrow();
+        return user.map(modelReadMapper::mapFrom);
+    }
+
+
+    @Transactional
+    public CreateOrderModel purchaseCertificate(long userId, long certificateId) {
+
+        Optional<ModelCertificate> certificate = certificateService.getEntity(certificateId);
+        certificate.get().setCreateDate(LocalDateTime.now());
+        certificateService.updateEntity(certificateId, (onlyCertificateMapper.mapFrom(certificate.get())));
+        Optional<User> user = Optional.ofNullable(repository.getEntityById(userId)).orElseThrow();
+        List<Certificate> list = new ArrayList<>();
+        Optional<ModelCertificate> updateCertificate = certificateService.getEntity(certificateId);
+        list.add(certificateMapper.mapFrom(updateCertificate.get()));
+        CreateOrderModel model = CreateOrderModel.builder()
+                .user(user.map(modelReadMapper::mapFrom).orElseThrow())
+                .certificate(onlyCertificateReadMapper.buildListCertificates(list)).build();
+        model.setUser((user.map(modelReadMapper::mapFrom).orElseThrow()));
+        model.setCertificate(onlyCertificateReadMapper.buildListCertificates(list));
+        model.setCost(updateCertificate.get().getPrice());
+        model.setDatePurchase(LocalDateTime.now());
+        orderRepository.saveOrder(orderMapper.mapFrom(model));
```

Comment:
Why don't you just create Order here? I see additional logic of creation CreateOrderModel

Reply:
brought order.